tests/e2e/specs/edit-posts.test.js

```diff
@@ -53,24 +49,24 @@ test.describe( 'Edit Posts', () => {
 		await editor.publishPost();
 		await admin.visitAdminPage( '/edit.php' );
 
-		await page.waitForSelector( '#the-list .type-post' );
+		const listTable = page.getByRole( 'table', { name: 'Table ordered by' } );
+		await expect( listTable ).toBeVisible();
 
 		// Click the post title (edit) link
-		await page.locator( `//a[contains(@class, "row-title")][contains(text(), "${ title }")]` ).click();
+		await listTable.getByRole( 'link', { name: `“${ title }” (Edit)` } ).click();
 
 		// Wait for the editor iframe to load, and switch to it as the active content frame.
-		const editorFrame = await page.waitForSelector( 'iframe[name="editor-canvas"]' );
-
-		const innerFrame = await editorFrame.contentFrame();
+		await page
+				.frameLocator( '[name=editor-canvas]' )
+				.locator( 'body > *' )
+				.first()
+				.waitFor();
 
-		// Wait for title field to render onscreen.
-		await innerFrame.waitForSelector( '.editor-post-title__input' );
+		const editorPostTitle = editor.canvas.getByRole( 'textbox', { name: 'Add title' } );
```

Comment:
This looks like an anti-pattern to me. Above, we explicitly wait for the `editor-canvas` frame to be ready via [`frameLocator()`](https://playwright.dev/docs/api/class-page#page-frame-locator), and here we reference canvas by the `canvas` object, [which is instantiated](https://github.com/WordPress/gutenberg/blob/537316ad5666c5574a4370c4f285e5f953315739/packages/e2e-test-utils-playwright/src/editor/index.ts#L40) by [`frame()`](https://playwright.dev/docs/api/class-page#page-frame).

Shouldn't the `canvas` object be instantiated by the `frameLocator` so we don't have to juggle like that? 

The only reason we need to add the `frameLocator...waitFor` step is that the `frame` locator does not support auto-waiting, so if the `iframe` element is not available at the moment of calling `frame`, the canvas will refer to the `page` object instead which will fail the test. 

cc: @kevin940726

Reply:
Selecting block editor elements via `editor.canvas.locator` in Gutenberg tests is standard practice. I'm trying to keep these similar.

Reply:
Yeah, that part should stay as is, only that the `editor.canvas` should be calling `frameLocator` instead of `frame`. I'll try spinning up a PR with the refactor to make it clearer.

Reply:
`frameLocator` for some reason doesn't allow some APIs like `evaluate`, which makes calling frame-level APIs a bit awkward. For instance, if we want to get the active element in the frame wme will then have to do `editor.canvas.locator('html').evaluate(() => document.activeElement.innerHTML)` etc. It's not that bad but we already have some existing code that relies on `editor.canvas` being able to call `evaluate`. I haven't found a way to properly mitigate that in gutenberg, but perhaps we can try that in core? But I guess consistency also matters here 😅 .

Reply:
Just pushed migrated tests that use the `frameLocator` API instead of `frame` in https://github.com/WordPress/gutenberg/pull/54911. It required a bit of juggling, but ultimately it wasn't that bad (assuming, of course, that the tests are going to pass! 😅). Let me know what you think! 🙌 

Re: evaluate from frameLocator

It looks like [the recommended approach](https://github.com/microsoft/playwright/issues/22924#issuecomment-1542498390) is to call 

```js
editor.canvas.locator( ':root' ).evaluate( /**/ );
```

Reply:
It's a bit more work than I anticipated (of course 🙄), but it's going well. The worst part to figure out so far was drag&drop via evaluated `DataTransfer`. TIL that the DataTransfer can't be transferred from one frame to another (e.g., from the main page to the editor canvas). It needs to be instantiated and dropped in the same frame. [Here's the commit that addressed that](https://github.com/WordPress/gutenberg/pull/54911/commits/51a3a7515598add637d705106782af994e398447).

Reply:
This is now ready to review! https://github.com/WordPress/gutenberg/pull/54911 cc @kevin940726 @swissspidy

Reply:
...and it's merged! 🎉